btm/src/main/java/bitronix/tm/resource/messaging/MessagingPooledConnection.java

```diff
@@ -0,0 +1,335 @@
+            poolingConnectionFactory.setUseTmJoin(true);
+        }
+        
+        this.jmxName = "bitronix.tm:type=JMS,UniqueName=" + ManagementRegistrar.makeValidName(poolingConnectionFactory.getUniqueName()) + ",Id=" + poolingConnectionFactory.incCreatedResourcesCounter();
+        ManagementRegistrar.register(jmxName, this);
+    }
+
+    public XAConnection getXAConnection() {
+        return xaConnection;
+    }
+
+    public PoolingConnectionFactory getPoolingConnectionFactory() {
+        return poolingConnectionFactory;
+    }
+
+    public synchronized RecoveryXAResourceHolder createRecoveryXAResourceHolder() throws JMSException {
+        DualSessionWrapper dualSessionWrapper = new DualSessionWrapper(this, false, 0);
+        dualSessionWrapper.getSession(true); // force creation of XASession to allow access to XAResource
+        return new RecoveryXAResourceHolder(dualSessionWrapper);
+    }
+
+    @Override
+    public synchronized void close() throws JMSException {
+        if (!closed) {
+            setState(STATE_CLOSED);
+            xaConnection.close();
+        }
+        closed = true;
+    }
+
+    @Override
+    public List<XAResourceHolder> getXAResourceHolders() {
+        sessionsReadLock.lock();
+        try {
+            return new ArrayList<XAResourceHolder>(sessions);
+        } finally {
+            sessionsReadLock.unlock();
+        }
+    }
+
+    @Override
+    public Object getConnectionHandle() throws Exception {
+        if (log.isDebugEnabled()) log.debug("getting connection handle from " + this);
+        int oldState = getState();
+
+        setState(STATE_ACCESSIBLE);
+
+        if (oldState == STATE_IN_POOL) {
+            if (log.isDebugEnabled()) log.debug("connection " + xaConnection + " was in state IN_POOL, testing it");
+            testXAConnection();
+        }
+        else {
+            if (log.isDebugEnabled()) log.debug("connection " + xaConnection + " was in state " + Decoder.decodeXAStatefulHolderState(oldState) + ", no need to test it");
+        }
+
+        if (log.isDebugEnabled()) log.debug("got connection handle from " + this);
+        return new MessagingConnectionHandle(this, xaConnection);
+    }
+
+    private void testXAConnection() throws JMSException {
+        if (!poolingConnectionFactory.getTestConnections()) {
+            if (log.isDebugEnabled()) log.debug("not testing connection of " + this);
+            return;
+        }
+
+        if (log.isDebugEnabled()) log.debug("testing connection of " + this);
+        XASession xaSession = xaConnection.createXASession();
+        try {
+            TemporaryQueue tq = xaSession.createTemporaryQueue();
+            tq.delete();
+        } finally {
+            xaSession.close();
+        }
+    }
+
+    protected void release() throws JMSException {
+        if (log.isDebugEnabled()) log.debug("releasing to pool " + this);
```

Comment:
braces / remove if + use formatting. 
There are similar ones to this in this class